selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RolloutUpdateStrategyWithEditorTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.hotupdate;
+
+import static org.eclipse.che.selenium.core.project.ProjectTemplates.MAVEN_SPRING;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.google.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.che.api.system.shared.SystemStatus;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.client.CheTestSystemClient;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.client.TestUserPreferencesServiceClient;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.webdriver.WebDriverWaitFactory;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class RolloutUpdateStrategyWithEditorTest {
+  private static final int TIMEOUT_FOR_ROLLING_UPDATE_FINISH = 100;
+  private static final int RESTORE_IDE_AFTER_REFRESH_TIMEOUT = 10;
+  private static final int EXPECTED_PREFERENCES_RESPONCE_CODE = 200;
+  private static final String PROJECT_NAME = "default-spring-project";
```

Comment:
May be we will use all HTTP statuses from single entry point. I propose from this: `Response.Status.ACCEPTED`

Reply:
Good catch!

Reply:
Even more - we doesn't need to check response code so as in case of error `HttpJsonResponse#request()` throws one of the next type of exception: ServerException, UnauthorizedException, ForbiddenException, NotFoundException, ConflictException, BadRequestException.
So, just calling `TestUserPreferencesServiceClient#getPreferences()` would be enough to check if everything is okay with WS master server.

Reply:
Totally agree